File: src/TableOfContents.java
import java.util.ArrayList;
import java.util.List;

public class TableOfContents {
    public static class Entry {
        public final String title;
        public final Integer page;

        public Entry(String title, Integer page) {
            this.title = title;
            this.page = page;
        }
    }

    protected final List<Entry> entries = new ArrayList<>();

    public void addEntry(String title) {
        addEntry(title, null);
    }

    public void addEntry(String title, Integer page) {
        entries.add(new Entry(title, page));
    }

    public void print() {
        System.out.println("TableOfContents");
        for (Entry e : entries) {
            if (e.page != null) {
                System.out.println(" - " + e.title + " .... " + e.page);
            } else {
                System.out.println(" - " + e.title);
            }
        }
    }
}

File: TableOfContents.js
class TableOfContents {
  constructor() {
    this.entries = [];
  }

  addEntry(title, page = null) {
    this.entries.push({ title, page });
  }

  // Print simplificat și moștenit de toate clasele
  print() {
    const kind = this.constructor.name;
    const label = this.name || this.title || this.text || this.imageName || '';
    console.log(label ? `${kind}: ${label}` : kind);
  }
}
